phantazm-mob/src/main/java/com/github/phantazmnetwork/mob/goal/FollowEntityGoal.java

```diff
@@ -0,0 +1,60 @@
+package com.github.phantazmnetwork.mob.goal;
+
+import com.github.phantazmnetwork.api.VecUtils;
+import com.github.phantazmnetwork.api.config.VariantSerializable;
+import com.github.phantazmnetwork.mob.PhantazmMob;
+import com.github.phantazmnetwork.mob.target.TargetSelector;
+import com.github.phantazmnetwork.neuron.bindings.minestom.entity.goal.NeuralGoal;
+import net.kyori.adventure.key.Key;
+import net.minestom.server.entity.Entity;
+import org.jetbrains.annotations.NotNull;
+
+import java.util.Objects;
+
+public record FollowEntityGoal<TEntity extends Entity>(@NotNull TargetSelector<TEntity> entitySelector) implements GoalCreator, VariantSerializable {
+
+    public final static Key SERIAL_KEY = Key.key("phantazm", "follow_entity_goal");
```

Comment:
on Mapdata branch, there is a class "Namespaces" that holds the phantazm namespace as a constant string